model/src/lib.rs

```diff
@@ -329,13 +332,75 @@ impl TokenPair {
     }
 }
 
+#[derive(Copy, Eq, PartialEq, Clone, Default)]
+pub struct DomainSeparator(pub [u8; 32]);
+
+impl std::str::FromStr for DomainSeparator {
+    type Err = FromHexError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self(FromHex::from_hex(s)?))
+    }
+}
+
+impl std::fmt::Debug for DomainSeparator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut hex = [0u8; 64];
+        // Unwrap because we know the length is correct.
+        hex::encode_to_slice(self.0, &mut hex).unwrap();
+        // Unwrap because we know it is valid utf8.
+        f.write_str(std::str::from_utf8(&hex).unwrap())
+    }
+}
+
+impl DomainSeparator {
+    pub fn get_domain_separator(chain_id: u64, contract_address: H160) -> Self {
+        lazy_static! {
+            /// The EIP-712 domain name used for computing the domain separator.
+            static ref DOMAIN_NAME: [u8; 32] = signing::keccak256(b"Gnosis Protocol");
+
+            /// The EIP-712 domain version used for computing the domain separator.
+            static ref DOMAIN_VERSION: [u8; 32] = signing::keccak256(b"v2");
+
+            /// The EIP-712 domain type used computing the domain separator.
+            static ref DOMAIN_TYPE_HASH: [u8; 32] = signing::keccak256(
+                b"EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)",
+            );
+        }
+        let abi_encode_string = encode(&[
+            Token::Uint((*DOMAIN_TYPE_HASH).into()),
+            Token::Uint((*DOMAIN_NAME).into()),
+            Token::Uint((*DOMAIN_VERSION).into()),
+            Token::Uint(chain_id.into()),
+            Token::Address(contract_address),
+        ]);
+
+        DomainSeparator(signing::keccak256(abi_encode_string.as_slice()))
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use chrono::NaiveDateTime;
     use serde_json::json;
     use std::str::FromStr;
 
+    #[test]
+
+    fn domain_separator_rinkeby() {
+        let contract_address: H160 = "91D6387ffbB74621625F39200d91a50386C9Ab15".parse().unwrap();
+        let chain_id: u64 = 4;
+        let domain_separator_rinkeby =
+            DomainSeparator::get_domain_separator(chain_id, contract_address);
+        // domain separator is taken from rinkeby deployment at address 91D6387ffbB74621625F39200d91a50386C9Ab15
+        let expected_domain_separator: DomainSeparator =
+            "9d7e07ef92761aa9453ae5ff25083a2b19764131b15295d3c7e89f1f1b8c67d9"
+                .parse()
+                .unwrap();
```

Comment:
Here too. And then I think we don't need the DomainSeparator FromStr implementation anymore.
```suggestion
        let expected_domain_separator: DomainSeparator = DomainSeparator(hex!(
            "9d7e07ef92761aa9453ae5ff25083a2b19764131b15295d3c7e89f1f1b8c67d9"
        ));
```